### DISCUSSION TITLE
Use Celery with microcontrollers / IoT Devices.

### DISCUSSION THREAD

**Discussion by lMengSun** [celery/celery#8593] [Ideas]
Hello all. I am thinking if Celery (worker) could be also implemented within a microcontroller. Nowadays if we think about STM32 or ESP32, those have been used to connecting to MQTT servers and RabbitMQ has also a port with MQTT. I was wondering if technically a STM32/ESP32 IoT devices could be used to connect to a MQTT-like broker, then receive tasks signatures, executing and send results back? From architecture point of view data flow would be like:

ESP32/STM32 IoT Devices <-> FreeRTOS <-> MQTT Libs <-> LwIP <-> RabbitMQ with MQTT-extension <-> "Rest of the network"
|________________________MCU Domain____________________|

This for sure implies on the MCU side, all worker codes will be implemented by C and sounds like a duplicated-implementation from its Python counterpart.

I am pretty new person who just touch Celery. So my idea / hypothesis might not be correct anyway... :)

**Comment by auvipy** [comment]
it is not clear what you actually wanted to share/imply
